src/visualization/types/Graph/view.tsx

```diff
@@ -192,20 +198,43 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
     const singleClickHandler = (
       plotInteraction: InteractionHandlerArguments
     ) => {
-      event('xyplot.annotations.create_annotation.show_overlay')
-      dispatch(
-        showOverlay(
-          'add-annotation',
-          {
-            createAnnotation,
-            startTime: plotInteraction.valueX,
-          },
-          () => {
-            event('xyplot.annotations.create_annotation.cancel')
-            dismissOverlay()
-          }
-        )
+      const annotationLayers: any =
+        config.layers.find(l => l.type === 'annotation') ?? []
+
+      const clickedAnnotation = annotationLayers.annotations?.find(
+        f =>
+          f.startValue === Math.round(plotInteraction.valueX as number) ||
+          f.stopValue === Math.round(plotInteraction.valueX as number)
       )
+
+      if (clickedAnnotation) {
+        event('xyplot.annotations.edit_annotation.show_overlay')
+        dispatch(
+          showOverlay(
+            'edit-annotation',
+            {editAnnotation, clickedAnnotation},
+            () => {
+              event('xyplot.annotations.edit_annotation.cancel')
+              dismissOverlay()
+            }
+          )
+        )
```

Comment:
i think this needs to be removed right? since we're changing how we launch the annotation popup

Reply:
Once we actually come up with that work and it gets merged in, then i would remove this. But as of  now, to get this working and unblock work on edit/delete, id leave this here so we can focus on those